internal/rukpakctl/ca.go

```diff
@@ -0,0 +1,26 @@
+package rukpakctl
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetClusterCA returns an x509.CertPool by reading the contents of a Kubernetes Secret. It uses the provided
+// client to get the requested secret and then loads the contents of the secret's "ca.crt" key into the cert pool.
+func GetClusterCA(ctx context.Context, cl client.Reader, ns, secretName string) (*x509.CertPool, error) {
```

Comment:
Would it be easier to pass a NamespacedName instead of individual ns/secretName string fields?